w3lib/url.py

```diff
@@ -204,8 +204,15 @@ def _add_or_replace_parameters(url, params):
     parsed = urlsplit(url)
     args = parse_qsl(parsed.query, keep_blank_values=True)
 
-    new_args = OrderedDict(args)
-    new_args.update(params)
+    new_args = []
+    changing_params = list(params.keys())
+    for name, value in args:
+        if name in params.keys():
```

Comment:
no need to get the list of `.keys()`, it should be possible to test like `if name in params` if `params` is a dict.